### PR TITLE
Fix vehicle lock pick

### PR DESCRIPTION
**Pull request by rube200**

### CODE REVIEW

File: unturned/OpenMod.Unturned/Vehicles/Events/VehicleEventsListener.cs [hunk 1]
```diff
@@ -471,41 +459,37 @@ public static bool PreVehicleLock(VehicleManager __instance, CSteamID steamID)
...
                 var vehicle = VehicleManager.findVehicleByNetInstanceID(instanceID);
-                if (vehicle == null)
+                if (vehicle is null)
                 {
                     return;
                 }
 
                 var nativePlayer = PlayerTool.getPlayer(owner);
-                if (nativePlayer == null)
+                if (nativePlayer is null)
```
Comment (Trojaner) [inline unturned/OpenMod.Unturned/Vehicles/Events/VehicleEventsListener.cs:485]: Please use == null instead so it checks if Unity has disposed / destroyed it

File: unturned/OpenMod.Unturned/Vehicles/Events/VehicleEventsListener.cs [hunk 2]
```diff
@@ -381,44 +382,38 @@ public static void PreRemovePlayer(InteractableVehicle __instance, byte seatInde
...
-                if (fromSeatIndex < __instance.passengers?.Length && toSeatIndex < __instance.passengers?.Length)
-                {
-                    var passenger = __instance.passengers[toSeatIndex];
+                if (__instance.passengers is null || fromSeatIndex >= __instance.passengers.Length || toSeatIndex >= __instance.passengers.Length)
+                    return;
 
-                    var player = passenger?.player?.player;
+                var passenger = __instance.passengers[toSeatIndex];
+                var player = passenger?.player?.player;
+                if (player is null)
```
Comment (Trojaner) [inline unturned/OpenMod.Unturned/Vehicles/Events/VehicleEventsListener.cs:413]: == null instead

File: unturned/OpenMod.Unturned/Vehicles/Events/VehicleEventsListener.cs [hunk 3]
```diff
@@ -471,41 +459,37 @@ public static bool PreVehicleLock(VehicleManager __instance, CSteamID steamID)
...
             }
 
             [HarmonyPatch(typeof(VehicleManager), "tellVehicleLock")]
             [HarmonyPostfix]
             [UsedImplicitly]
-            public static void PostVehicleLock(VehicleManager __instance, uint instanceID, CSteamID owner, CSteamID group, bool locked)
+            public static void PostVehicleLock(uint instanceID, CSteamID owner, CSteamID group, bool locked)
             {
                 var vehicle = VehicleManager.findVehicleByNetInstanceID(instanceID);
-                if (vehicle == null)
```
Comment (Trojaner) [inline unturned/OpenMod.Unturned/Vehicles/Events/VehicleEventsListener.cs:493]: == null instead

**Review by Trojaner — CHANGES_REQUESTED** [review]

**Comment by IAmSilK** [comment]
@rube200 could you also change `IsCancelled = allow` to `IsCancelled = !allow`. This is a bug as well.

**Review by Trojaner — APPROVED** [review]

File: unturned/OpenMod.Unturned/Vehicles/Events/VehicleEventsListener.cs [hunk 4]
```diff
@@ -135,7 +135,7 @@ private void OnVehicleLockpicked(InteractableVehicle vehicle, Player instigating
 
             var @event = new UnturnedVehicleLockpickingEvent(new UnturnedVehicle(vehicle), instigator!)
             {
-                IsCancelled = allow
+                IsCancelled = !allow
```
Comment (rube200) [inline unturned/OpenMod.Unturned/Vehicles/Events/VehicleEventsListener.cs:138]: @IAmSilK here

**Comment by rube200** [comment]
@IAmSilK That was the hole  point of doing this changes/PR

**Pull request merged**